internal/dag/dag.go

```diff
@@ -371,10 +372,7 @@ func (t *TCPProxy) Visit(f func(Vertex)) {
 
 // Service represents a single Kubernetes' Service's Port.
 type Service struct {
-	Name      string
-	Namespace string
-
-	ServicePort v1.ServicePort
+	Service WeightedService
```

Comment:
Do we need `WeightedService` here? Could just use `Service` and remove the stutter of `service.service.xxx`.

Reply:
Originally, I wanted to use a full ServiceCluster, but that ended up being a bit too much refactoring for now.

Reply:
Is there a reason to not just remove the extra struct? I think it would read much better.

Reply:
Keeping the struct improves readability and discoverability. It's clear to the reader that a WeighedService is being used, whereas if the fields were embedded directly it would be nard to find.

Reply:
Sure but that only makes sense if you're going to have another type of service. If you have a service with no weights, the same "weighted" struct is still used, so I don't think it's any clearer. 

If wanting to call out the specifics of the weighted service then it should be renamed to `Weighted` or something so you'd call it as `Service.Weighted` since calling it `Service.Service` doesn't make it any clearer in my eye. 

I'd just remove the extra struct, but ¯\_(ツ)_/¯

Reply:
I can rename to `Weighted`.

Reply:
I agree with renaming to Weighted.

Reply:
Renamed.